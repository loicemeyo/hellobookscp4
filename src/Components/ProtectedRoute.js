import React from "react";
import { Route, Redirect } from "react-router";

/**
 * This Component protect urls so that they cannot be accessed before login
 */

const ProtectedRoute = ({ component: Values, Component, ...rest}) => (
  <Route
    {...rest}
    render={props => (
      window.localStorage.getItem("access_token")?
        (<Values{...props}/>):
        (<Redirect to={{pathname: "/login"}}/>)
    )}
  />
);
export default ProtectedRoute;