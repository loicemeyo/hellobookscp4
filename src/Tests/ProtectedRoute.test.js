import React from "react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";
import ProtectedRoute from "../Components/ProtectedRoute";

describe("Test ProtectedRoute Component", () => {
    const wrapper = shallow(<ProtectedRoute />);

    it("ProtectedRoute component renders without crushing", () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });

});