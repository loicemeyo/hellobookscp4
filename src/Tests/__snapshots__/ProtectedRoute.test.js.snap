// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProtectedRoute Component ProtectedRoute component renders without crushing 1`] = `
<Route
  render={[Function]}
/>
`;
